climada/util/coordinates.py

```diff
@@ -1155,23 +1158,81 @@ def get_admin1_info(country_names):
         Shape according to Natural Earth.
     """
 
-    if isinstance(country_names, str):
-        country_names = [country_names]
+    if isinstance(countries, (str, int, float)):
+        countries = [countries]
+    if not isinstance(countries, list):
+        raise TypeError("Invalid type for input parameter 'countries'")
     admin1_file = shapereader.natural_earth(resolution='10m',
                                             category='cultural',
                                             name='admin_1_states_provinces')
     admin1_recs = shapefile.Reader(admin1_file)
     admin1_info = dict()
     admin1_shapes = dict()
-    for iso3 in country_names:
-        admin1_info[iso3] = list()
-        admin1_shapes[iso3] = list()
+    for country in countries:
+        if isinstance(country, (int, float)):
+            country = f'{int(country):03d}' # transform numerric code to str
+        country = pycountry.countries.lookup(country).alpha_3 # get iso3a code
+        admin1_info[country] = list()
+        admin1_shapes[country] = list()
         for rec, rec_shp in zip(admin1_recs.records(), admin1_recs.shapes()):
-            if rec['adm0_a3'] == iso3:
-                admin1_info[iso3].append(rec)
-                admin1_shapes[iso3].append(rec_shp)
+            if rec['adm0_a3'] == country:
+                admin1_info[country].append(rec)
+                admin1_shapes[country].append(rec_shp)
```

Comment:
There's a chance of a confusing error here if the ISO codes in pycountry are different from those in Natural Earth (which could happen with different versions of the packages). We should check that every country that pycountry identifies is successfully extracted from Natural Earth.
e.g. add a check that every list in `admin1_info` has length > 0

Reply:
There's also the edge case to worry about, where a user provides duplicate countries in their list. Do we care about this?

My understanding right now is that `get_admin1_info(['CHE', 'CHE'])` returns the same as `get_admin1_info('CHE')` (but does the calculations twice). This seems like safe behaviour to be honest, but I wanted to check.